lib/util/utils.ts

```diff
@@ -13,6 +13,29 @@ import { Constants } from "./constants";
  */
 export const isNode = typeof navigator === "undefined" && typeof process !== "undefined";
 
+/**
+ * Adapts a WithHttpOperationResponse method to unwrap the body and accept an optional callback.
+ * @param httpOperationResponseMethod the method to call and apply a callback to
+ * @param args the arguments to the method, optionally including a trailing callback.
+ */
+export function responseToBody(httpOperationResponseMethod: (...args: any[]) => Promise<HttpOperationResponse>, ...args: any[]): Promise<any> | undefined {
+  // The generated code will always pass (options, callback) as the last two args.
+  // But `options` could actually be the callback for the sake of user convenience.
+  let callback = args.pop();
+  if (typeof callback !== "function" && typeof args[args.length - 1] === "function") {
+    callback = args.pop();
+  }
```

Comment:
I need a little bit of explanation here. I know what should be happening (remove callback from the end of the argument list, if it is there). However, doesn't `pop()` remove the last argument from the list? What if no arguments are provided? What if the only argument is a value destined for the `httpOperationResponseMethod`? Won't `pop()` remove that argument so that `httpOperationResponseMethod` won't be provided the argument that it should be?
Also, I'm confused by line 25. If the last argument in the `args` array wasn't a callback function, but the second-to-last argument (in the original unmodified array) is, then we call `pop()` again? Doesn't your comment explain that there should never be an argument after the callback function?